articles/112_generated_interfaces_cpp.md

```diff
@@ -103,7 +103,19 @@ All other constants are declared as `static const` members in the struct and the
 
 ### Constructors
 
-The *default constructor* initializes all members with their default value.
+The *default constructor* initializes all members with their default value; if a field doesn't have a default value, then the field is [value-initialized](http://en.cppreference.com/w/cpp/language/value_initialization).
+In some cases this may not be desirable, since these fields will often be immediately overwritten with user-provided values.
+Therefore, the constructor takes an optional directive of type `enum rosidl_runtime_cpp_msg_init_type` to control how initialization is done:
```

Comment:
I still don't see why a C++ specific type should have such a long type name (`rosidl_runtime_cpp_msg_init_type`). Same for the enums below.

Reply:
Yeah, I agree it is a bit unwieldy, though I think you do want to encapsulate both a prefix and a description.  How about `rosidl_runtime_cpp_msg_init`?  Or `rosidl_msg_init_type`? Or `rosidl_msg_init`?

As for the name of the enums themselves, that comes down to whether we use the C++-specific enum class or C-style enum.  If we use the latter, then they go into the global namespace, so you want something unique.  If we use the former, then we can get rid of a lot of that prefix.  I think this is the one unresolved question here; do we stick with C-style enums or go with C++-style ones?  C-style enums will be better for maintenance across C and C++ longterm, but C++ ones are easier to work with.  Further thoughts?

Reply:
For a C++ interface I expect C++ enum class (as proposed in https://github.com/ros2/design/pull/136#discussion_r134885656). Internally the enum values can be defined based on the C enum values - no need to duplicate the constants.